### PR TITLE
9049 validate NR only when IA filing is in DRAFT or PENDING status

### PR DESCRIPTION
**Pull request by vysakh-menon-aot**
*Issue #:* /bcgov/entity#9049

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the business-filings-ui license (Apache 2.0).

### CODE REVIEW

File: src/App.vue [hunk 1]
```diff
@@ -593,26 +593,28 @@ export default {
...
     },
 
-    storeNrData (nr: any): void {
-      // check if NR is valid
-      if (!this.isNrValid(nr)) {
-        this.nameRequestInvalidDialog = true
-        throw new Error('Invalid NR data')
-      }
+    storeNrData (nr: any, ia: any): void {
+      if (ia.filing.header.status in [FilingStatus.DRAFT, FilingStatus.PENDING]) {
```
Comment (severinbeauvais) [inline src/App.vue:597]: I'm trying to think what states the filing could be in, and what should be done in various cases...
Here, you've determined that we should check the NR only if the filing is Draft or Pending. What about the other states?
A couple of options to make this code "fuller": add a comment explaining the business rule here; or use a switch-case (with comments) to itemize what happens for all states. Or ...?

Comment (severinbeauvais) [reply src/App.vue:597]: Or change this condition to "don't check the NR if the filing is Completed"?

What do you think?

Comment (vysakh-menon-aot) [reply src/App.vue:597]: I think it should be fine to check !Completed. Will change it

**Review by severinbeauvais — APPROVED** [review]
Looks good to me!

**Pull request merged**